mongo/datadog_checks/mongo/mongo.py

```diff
@@ -25,6 +25,21 @@
 ALLOWED_CUSTOM_QUERIES_COMMANDS = ['aggregate', 'count', 'find']
 
 
+def build_url(scheme, host, path='/', username=None, password=None, query_params=None):
+    # type: (str, str, str, str, str, dict) -> str
+    """Build an URL from individual parts. Makes sure that parts are properly URL-encoded."""
+    if username and password:
+        netloc = '{}:{}@{}'.format(quote_plus(username), quote_plus(password), host)
+    else:
+        netloc = host
+
+    path_params = ""
+    query = urlencode(query_params or {})
+    fragment = ""
+
+    return urlunparse([scheme, netloc, path, path_params, query, fragment])
```

Comment:
```suggestion
    return urlunparse((scheme, netloc, path, path_params, query, fragment))
```

Shouldn't change much but, from the doc `Construct a URL from a tuple as returned by urlparse(). `: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlunparse

Reply:
The term `tuple` there comes from the fact that `urlparse` returns a tuple-like object. The next sentence reads:

> The `parts` argument can be any six-item iterable.

So I think we're fine with a list :)